docs/enhancements/multi-rhel-fips.md

```diff
@@ -0,0 +1,137 @@
+a dynamically linked `openshift-install` binary against the openssl libraries
+present on our container image. Today this is not a problem because all
+`openshift-install` binaries in use have been expecting to link to RHEL 8 based
+openssl libraries, but now OpenShift 4.16 will ship and installer that requires
+RHEL 9 libraries.
+
+This will require assisted-service to maintain a way to run the
+`openshift-install` binary in a compatible environment for multiple openssl
+versions.
+
+## Motivation
+
+FIPS compliance is important for our customers and assisted-service should be
+able to install FIPS compliant clusters.
+
+### Goals
+
+- Allow for a single installation of assisted-service to install FIPS-compliant
+  clusters using `openshift-install` binaries built against RHEL 8 or RHEL 9
+
+- Allow for FIPS compliant clusters to be installed from the SaaS offering or
+  the on-prem offering
+
+### Non-Goals
+
+- Changing cluster install interfaces to accommodate new FIPS requirements
+  should be avoided
+
+- Dynamically determining a given release's RHEL version. The RHEL version will
+  be a part of configuration tied to the minor release
+
+## Proposal
+
+Two additional containers will run alongside assisted-service in the same pod.
+These "installer-runner" containers will, when triggered by assisted-service,
+run a specified installer binary to generate the manifests required for a
+particular install. These manifests will then be uploaded to whatever storage
+is in use for this deployment (local for on-prem, or s3 for SaaS) and
+assisted-service will take over as usual from there.
+
+Each of these containers should maintain its own cache of installers as
+assisted-service does now and should be built with the required packages to run
+the installer in FIPS compliance mode.
+
+### User Stories
+
+#### Story 1
+
+As a Cluster Creator, I want to install FIPS compliant clusters for any supported OpenShift version
+
+### Implementation Details/Notes/Constraints
+
+#### New Images
+
+Two new container images will need to be built, and published for every release
+footprint we support. These images will be created based on existing
+assisted-service code, but could be split into their own independent projects
+later.
+
+### Risks and Mitigations
+
+Shipping a new image is a non-trivial process. This may take more time than we
+have to set up. We could likely get away with using the existing
+assisted-service image with a different entrypoint for one of the runner images,
+but that still requires us to publish a new one for the architecture
+assisted-service will not be using.
+
+## Design Details [optional]
+
+The new runner containers will expose a HTTP server listening only for local
+connections. assisted-service will POST to this server when it needs manifests
+generated. The runner container will respond with any error that occurred while
+generating the manifests.
+
+### Open Questions
```

Comment:
I briefly called this out in `New Images` under `Implementation Details`, but if you think it would be good to put somewhere else or just talked about more prominently then we can do that.

I don't think it's an open question though. For this pass I'm going to create the service in `assisted-service` using the existing code to run the installer. Extracting and splitting everything up would definitely come later.